lib/filter.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+//////////////////////////////
+// contains, is, isNot, doesNotContain, exists, matches
+//
+// Include
+//////////////////////////////
+const addModule = function (results, pkg) {
+  try {
+    results[pkg.package.name] = require(pkg.package.name);
+  }
+  catch (e) {
+    if (!results.hasOwnProperty('__unrequireable')) {
+      results.__unrequireable = [];
+    }
+    results.__unrequireable.push(pkg.package.name);
+  }
+  return results;
+}
+
+module.exports = function ModuleFilter(packages, key) {
+  /**
+    * Contains
+  **/
+  this.containsSync = (search) => {
+    var results = {};
```

Comment:
I understand why you want to use `.map`, but the way in which you're expecting it to be used runs against what the required output of this function is (a map with keys/values, not an array)